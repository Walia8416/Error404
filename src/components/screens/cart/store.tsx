import {
  View,
  Text,
  TouchableOpacity,
  ScrollView,
  StyleSheet,
} from 'react-native';

import {Colors} from '../../../constants/colors';
import Header from '../../helpers/header/Header';
import React, {useEffect, useRef, useState} from 'react';
import {RouteStackParamList} from '../../../Routes';
import {Styles} from '../../../styles/homescreen';
import {SafeAreaView} from 'react-native-safe-area-context';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {Bold, RobMono} from '../../../constants/Fonts';
import {RFValue} from 'react-native-responsive-fontsize';
import {Screen_Height} from '../../../constants/constants';
import {getorders} from '../../../store/actions/getOrders';
import {useAppSelector} from '../../../store/store';
import {useDispatch} from 'react-redux';
import OrderCard from './orderCard';
import Loading from '../home/helper/Loading';

const Store: React.FC<RouteStackParamList<'Store'>> = ({
  navigation,
}: RouteStackParamList<'Store'>) => {
  const [shadow, setShadow] = React.useState(false);
  const {orders} = useAppSelector(state => state.orders);
  const [current, setCurrent] = useState([]);
  const dispatch = useDispatch();

  useEffect(() => {
    setCurrent(orders.orders);
  }, [orders]);

  if (current && current?.length > 0) {
    return (
      <SafeAreaView style={{backgroundColor: Colors.white, flex: 1}}>
        <Header
          testID={'menu'}
          navigation={navigation}
          scroll={shadow}
          creds={true}
          title={'Order History'}
        />
        <View style={styles.mainCon}>
          <ScrollView>
            {current.map(item => (
              <TouchableOpacity
                onPress={() =>
                  navigation.navigate('PdfView', {
                    f: `/storage/emulated/0/Android/data/com.bettervitty/files/${item.invoiceId}.pdf`,
                  })
                }>
                <OrderCard item={item} />
              </TouchableOpacity>
            ))}
          </ScrollView>
        </View>
      </SafeAreaView>
    );
  }
  return (
    <View style={{flex: 1}}>
      <Loading />
    </View>
  );
};

export default Store;
const styles = StyleSheet.create({
  mainCon: {
    backgroundColor: 'white',
    flex: 1,
    padding: 20,
  },

  headText: {
    marginBottom: 20,
    fontSize: RFValue(25),
    color: Colors.black,
    fontFamily: Bold,
  },
});
