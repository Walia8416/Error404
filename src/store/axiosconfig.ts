import axios from 'axios';
import AsyncStorage from '@react-native-async-storage/async-storage';

export const axiosInstance = axios.create({
  baseURL: 'http://3.108.203.2:8000/api',
});
