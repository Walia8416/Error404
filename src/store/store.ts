import {TypedUseSelectorHook, useDispatch, useSelector} from 'react-redux';
import {combineReducers, configureStore} from '@reduxjs/toolkit';
import {persistStore, persistReducer} from 'redux-persist';
import {storesSlice} from './reducers/stores';
import {productSlice} from './reducers/products';
import {orderSlice} from './reducers/getOrders';
import AsyncStorage from '@react-native-async-storage/async-storage';
const rootReducer = combineReducers({
  stores: storesSlice.reducer,
  products: productSlice.reducer,
  orders: orderSlice.reducer,
});

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};
const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      immutableCheck: false,
      serializableCheck: false,
    }),
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
