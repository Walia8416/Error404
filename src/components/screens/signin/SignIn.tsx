import {Alert, Image, Text, TouchableOpacity, View} from 'react-native';
import {heighttodp, widthtodp} from '../../../constants/Dimenstions';
import {Colors} from '../../../constants/colors';
import Images from '../../../constants/icon';
import InputFeild from '../../helpers/inputfields/InputField';
import AsyncStorage from '@react-native-async-storage/async-storage';
import React, {useEffect} from 'react';
import {RouteStackParamList} from '../../../Routes';
import {StyledText} from '../../../styled_components/styledtexts';
import {useForm} from 'react-hook-form';
import SignUpHeading from '../../helpers/SignUpHeading';
import styles from './SignInStyles';
import {login, loginUser} from '../../../store/actions/login';
import {useAppDispatch} from '../../../store/store';
import {getStores} from '../../../store/actions/stores';
import {getorders} from '../../../store/actions/getOrders';

const SignIn: React.FC<RouteStackParamList<'SignIn'>> = ({
  navigation,
}: RouteStackParamList<'SignIn'>) => {
  const {
    control,
    handleSubmit,
    formState: {errors, isValid},
  } = useForm({mode: 'onChange', shouldFocusError: true});
  const [check, setCheck] = React.useState(false);
  const [loading, setLoading] = React.useState(false);
  const dispatch = useAppDispatch();

  const storeData = async (keyd, value) => {
    try {
      await AsyncStorage.setItem(keyd, value);
    } catch (e) {
      console.log('fail save');
    }
  };

  const onSubmit = async (data: {Email: string; Password: string}) => {
    login({
      email: data.Email,
      password: data.Password,
    })
      .then(result => {
        if (result.status == 200) {
          dispatch(getStores(result.data.operator._id));
          dispatch(getorders(result.data.operator._id));
          AsyncStorage.setItem('Users', result.data.operator._id);
          storeData('Tokens', result.data.token);
          navigation.navigate('Home');
        } else {
          Alert.alert('Invalid Email or Password!');
        }
      })
      .catch(err => {
        console.error(err);
      });
  };

  return (
    <View style={styles.main}>
      <View
        style={{
          marginLeft: widthtodp(20),
          marginTop: heighttodp(30),
          marginBottom: heighttodp(20),
        }}>
        <SignUpHeading heading="Sign In" subheading="For " />
      </View>
      <View style={styles.inputText}>
        <InputFeild
          height={heighttodp(46)}
          width={widthtodp(360 - 48)}
          name="Email"
          text={'Email'}
          placeholder={'Enter Email Address'}
          placeholdertextcolor={Colors.placeHolder}
          controls={control}
          onError={errors.Username}
          errorText={errors.Username?.message}
          rules={{
            required: {
              value: true,
              message: 'Enter email address',
            },
          }}
        />
      </View>
      <View style={[styles.inputText, {marginTop: heighttodp(10)}]}>
        <InputFeild
          height={heighttodp(46)}
          width={widthtodp(360 - 48)}
          name="Password"
          text={'Password'}
          placeholder={'Password'}
          placeholdertextcolor={Colors.placeHolder}
          controls={control}
          onError={errors.Password}
          errorText={errors.Password?.message}
          secure={!check}
          rules={{
            required: {
              value: true,
              message: 'Enter Password',
            },
            minLength: {
              value: 2,
              message: 'Enter',
            },
          }}
        />
      </View>

      <View style={styles.login}>
        <TouchableOpacity
          disabled={!isValid}
          onPress={handleSubmit(onSubmit)}
          style={[
            styles.loginBtn,
            {backgroundColor: isValid ? Colors.x : Colors.inputBorder},
          ]}>
          <Text style={styles.loginText}>Log In</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.line} />

      <View
        style={{
          flexGrow: 1,
          justifyContent: 'flex-end',
          marginBottom: heighttodp(20),
        }}></View>
    </View>
  );
};

export default SignIn;
