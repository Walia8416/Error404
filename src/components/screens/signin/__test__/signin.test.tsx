import React from 'react';
import SignIn from '../SignIn';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../store/store';


const CreateTestProps = (props: Object) => ({
  navigation: {
    state: { params: {} },
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn(),
  },
  ...props,
});
let props: any;

describe('Sign in render', () => {
  props = CreateTestProps({});
  it('initial render test', () => {
    const signin = create(<Provider store={store}>
      <SignIn {...props} route={undefined} />
    </Provider>);
    expect(signin).toMatchSnapshot();
    // console.log(signin.root);
  });
});
