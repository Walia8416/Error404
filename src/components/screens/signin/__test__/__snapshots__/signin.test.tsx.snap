// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in render initial render test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": 1334,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 30,
        "marginLeft": 20,
        "marginTop": 40,
      }
    }
  >
    <View
      style={
        Object {
          "paddingBottom": 20,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#e40678",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 22,
              "marginBottom": 10,
              "marginTop": 15,
            },
          ]
        }
      >
        Sign In
      </Text>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#415f8b",
                "fontSize": 14,
                "marginBottom": 10,
              },
            ]
          }
        >
          One Stop for Everything Marketing
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 167,
        "justifyContent": "center",
        "marginHorizontal": 50,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "height": 96,
          "justifyContent": "center",
          "width": 650,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#839fc0",
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
      >
        Email
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Enter Email Address"
        placeholderTextColor="#839fc0"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#c4d7e7",
            "borderRadius": 0,
            "borderWidth": 1,
            "color": "#153059",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "height": 96,
            "paddingLeft": 16,
            "width": 650,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "flexDirection": "row",
            "height": 18,
            "marginTop": 12,
            "width": 210,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 167,
          "justifyContent": "center",
          "marginHorizontal": 50,
        },
        Object {
          "marginTop": 21,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "height": 96,
          "justifyContent": "center",
          "width": 650,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "#839fc0",
            "fontFamily": "Inter-Regular",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
      >
        Password
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Password"
        placeholderTextColor="#839fc0"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderColor": "#c4d7e7",
            "borderRadius": 0,
            "borderWidth": 1,
            "color": "#153059",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "height": 96,
            "paddingLeft": 16,
            "width": 650,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "flexDirection": "row",
            "height": 18,
            "marginTop": 12,
            "width": 210,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": 33.5,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 33.5,
              "width": 33.5,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 37.5,
        "justifyContent": "space-between",
        "marginHorizontal": 50,
        "marginTop": 16.5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#C4D7E7",
              "borderWidth": 1,
              "height": 16,
              "justifyContent": "center",
              "marginRight": 0,
              "width": 16,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 16,
                "opacity": 1,
                "width": 16,
              }
            }
            testID="checkboxfalse"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginLeft": 16.5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#415F8B",
              "fontFamily": "Inter-Regular",
              "fontSize": 24,
              "textAlignVertical": "center",
            }
          }
        >
          Remember me
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2886dc",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 24,
            "textAlignVertical": "center",
          }
        }
      >
        Forgot password?
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "justifyContent": "center",
        "marginTop": 62.5,
        "paddingHorizontal": 50,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#c4d7e7",
          "borderRadius": 100,
          "height": 100,
          "justifyContent": "center",
          "opacity": 1,
          "width": 650,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Inter-Medium",
            "fontSize": 31,
            "textAlign": "center",
          }
        }
      >
        Sign in
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#e8eff5",
        "height": 1,
        "marginVertical": 25,
        "width": 650,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 125,
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "borderColor": "#c4d7e7",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 88.93333333333334,
          "justifyContent": "center",
          "opacity": 1,
          "width": 312.5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#415F8B",
            "fontFamily": "Inter-Medium",
            "fontSize": 27,
          }
        }
      >
        Sign In using
      </Text>
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 35.5,
            "marginLeft": 16.5,
            "width": 33.5,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "borderColor": "#c4d7e7",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 88.93333333333334,
          "justifyContent": "center",
          "opacity": 1,
          "width": 312.5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#415F8B",
            "fontFamily": "Inter-Medium",
            "fontSize": 27,
          }
        }
      >
        Sign In using
      </Text>
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 41.5,
            "marginLeft": 16.5,
            "width": 41.5,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "row",
        "height": 56.5,
        "justifyContent": "center",
        "marginTop": 37.5,
        "width": 458.5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#33556a",
          "fontFamily": "Inter-Regular",
          "fontSize": 27,
        }
      }
    >
      Don’t have an account? 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2886dc",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
          }
        }
      >
        Sign Up
      </Text>
    </View>
  </View>
</View>
`;
